src/components/Resource/ResourceForm.tsx

```diff
@@ -404,6 +404,7 @@ export default function ResourceForm({ facilityId, id }: ResourceProps) {
                       <FormControl>
                         <div data-cy="select-assigned-user">
                           <UserSelector
+                            facilityId={form.watch("assigned_facility")?.id}
```

Comment:
as mentioned in the issue, it should list both users associated to the facility associated to the resource request and as well as current facility of the logged in user!